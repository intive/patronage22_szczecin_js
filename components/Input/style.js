import styled from 'styled-components'

export const InputComponent = styled.input`
  margin: 1rem 0;
  padding: 0.75rem 1.5rem 0.75rem 1rem;
  height: 3rem;
  border-radius: 8px;
  background: #F7F7F7;
  border: none;
  outline: none;

  &:focus {
    background: #E1E1E1;
  }
`
