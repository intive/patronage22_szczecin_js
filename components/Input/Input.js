import { InputComponent } from './style'
import CharacterCounter from '../CharacterCounter/CharacterCounter'

export default function Input (props) {
  return (
    <>
      <InputComponent {...props} />
      {props.maxLength ? <CharacterCounter inputValue={props.value} maxCharacters={props.maxLength} /> : null}
    </>
  )
}
