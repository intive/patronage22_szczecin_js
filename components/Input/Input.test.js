/* globals describe, it, expect  */

import { render, screen } from '@testing-library/react'
import Input from './Input'

describe('Input', () => {
  it('should render Input correctly', () => {
    const { container } = render(<Input />)

    expect(container).toMatchSnapshot()
  })

  it('should render Input with type correctly', () => {
    const { container } = render(<Input type='password' />)

    expect(container).toMatchSnapshot()
  })

  it('should render Input component and CharacterCounter with text 0/8', () => {
    render(<Input maxLength='8' />)

    expect(screen.getByText('0/8')).toBeTruthy()
  })
})
