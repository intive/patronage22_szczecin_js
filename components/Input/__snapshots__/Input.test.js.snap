// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render Input correctly 1`] = `
<div>
  <input
    class="sc-bdvvtL jwGgZq"
  />
</div>
`;

exports[`Input should render Input with type correctly 1`] = `
<div>
  <input
    class="sc-bdvvtL jwGgZq"
    type="password"
  />
</div>
`;
